### PR TITLE
Lesson03

### PR DESCRIPTION
**Pull request by yarikkk13**

### CODE REVIEW

File: src/app/components/section/section.component.ts [hunk 1]
```diff
@@ -32,8 +35,13 @@ export class SectionComponent implements OnInit {
     console.log(text);
   }
 
-  public showTheSearching(searchText: string): void {
+  public showTheSearching(searchText: string): any {
```
Comment (AndriyLeskiv) [inline src/app/components/section/section.component.ts:38]: add type ICourse

File: src/app/components/section/section.component.ts [hunk 2]
```diff
@@ -32,8 +35,13 @@ export class SectionComponent implements OnInit {
...
     console.log(searchText)
+
+    if (!searchText) {
+      return this.courses = this.courseService.getCourses()
+    }
+    return this.courses = this.courses?.filter(course => course.title.includes(searchText))
```
Comment (AndriyLeskiv) [inline src/app/components/section/section.component.ts:44]: why you filter data if you use filter pipe?

File: src/app/directives/creation-date.directive.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  constructor(private elementRef: ElementRef) {
+    const courseDate = this.directivesCourse?.creationTime.getTime();
+    const currentDate = Date.now();
+    const fourteenDays = 1209600000;
+
+    if (courseDate < currentDate && courseDate >= currentDate - fourteenDays) {
+      this.elementRef.nativeElement.style.border = "solid 1px green";
+    } else if (courseDate > currentDate) {
+      this.elementRef.nativeElement.style.border = "solid 1px blue";
+    }
```
Comment (AndriyLeskiv) [inline src/app/directives/creation-date.directive.ts:20]: move to ngOnInit()

File: src/app/components/section/section.component.html [hunk 4]
```diff
@@ -12,8 +12,14 @@
...
+  <div *ngIf="!courses?.length;else other"><p class="bg-white d-flex justify-content-around">{{ noData }}</p></div>
+  <ng-template #other>
+    <div>
+      <app-page (eventOfEditing)="edit($event)" [course]="c"
+                *ngFor="let c of courses | filter:searchText| orderBy; trackBy: trackByIndex"
+                (eventOfDeleting)="delete($event)"></app-page>
+    </div>
+  </ng-template>
   <div class="bg-white d-flex justify-content-center" style="margin-top: 15px">
     <a href="#" (click)="showMore()">{{ load }}</a>
```
Comment (AndriyLeskiv) [inline src/app/components/section/section.component.html:24]: image
where 'no data' message, why showMore exist?

File: src/app/directives/creation-date.directive.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  selector: '[appCreationDate]'
+})
+export class CreationDateDirective {
+  @Input()
+  directivesCourse: ICourse;
+
+  constructor(private elementRef: ElementRef) {
+    const courseDate = this.directivesCourse?.creationTime.getTime();
+    const currentDate = Date.now();
+    const fourteenDays = 1209600000;
```
Comment (AndriyLeskiv) [inline src/app/directives/creation-date.directive.ts:14]: where types? const fourteenDays: number

Comment (yarikkk13) [reply src/app/directives/creation-date.directive.ts]: done

File: src/app/components/section/section.component.ts (hunk 2 above)
Comment (yarikkk13) [reply src/app/components/section/section.component.ts:44]: cause if i dont use filter data and don't click button, show more doesn't disappear and noCourseFeelFreeToAddData doesn't appear

**Pull request merged**